src/web_interface/static/js/system_health.js

```diff
@@ -0,0 +1,160 @@
+const BOOTSTRAP_DANGER_COLOR = "#dc3545";
+const BOOTSTRAP_PRIMARY_COLOR = "#007bff";
+
+function change_button(button_id) {
+    element = document.getElementById(button_id);
+    if (element.classList.contains("fa-caret-down")) {
+        element.classList.remove("fa-caret-down");
+        element.classList.add("fa-caret-up");
+    } else {
+        element.classList.remove("fa-caret-up");
+        element.classList.add("fa-caret-down");
+    }
+}
+
+async function getSystemHealthData() {
+    const response = await fetch("/ajax/system_health");
+    return response.json();
+}
+
+async function updateSystemHealth() {
+    getSystemHealthData().then(data => data.systemHealth.map(entry => {
+        const statusElement = document.getElementById(`${entry._id}-status`);
+        statusElement.innerText = entry.status;
+        if (entry.status == "offline") {
+            statusElement.classList.add('text-danger');
+            statusElement.classList.remove('text-success');
+        } else {
+            statusElement.classList.remove('text-danger');
+            statusElement.classList.add('text-success');
+            document.getElementById(`${entry._id}-os`).innerText = entry.platform.os;
```

Comment:
We could reduce code depth here.

```suggestion
        if (entry.status == "offline") {
            statusElement.classList.add('text-danger');
            statusElement.classList.remove('text-success');
            return;
        }
        classList.remove('text-danger');
        classList.add('text-success');
        getElementById(`${entry._id}-os`).innerText = entry.platform.os;
```

Reply:
changed